static_inventories/betabarrel_cluster.yml

```diff
@@ -270,4 +270,7 @@ betabarrel_cluster:
     data_transfer:
     docs:
     smb_server:
+production:
```

Comment:
I am a bit confused. I think it complicates the whole static **inventory** structure just to label system as `production` or `development`. It usually goes for entire cluster that we want to use: for example, we don't mix and match development/production within the same cluster.

Wouldn't it be much simpler to just add variable (for example) called `production: true` or `production: false` in the `vars.yml`? Or perhaps: `cluster_type: development` / `production`?

Reply:
This one is up for debate; using inventory groups is the "_ansible way_" and separating development from testing is one of the examples from the [documentation for working with inventories](https://docs.ansible.com/ansible/latest/inventory_guide/intro_inventory.html). (In the documentation they use categories `prod` en `test`.) Using the inventory you can have:
```group_vars/production.yml```
with
```
some_variable: 'production_value'
```
and
```group_vars/development.yml```
with 
```
some_variable: 'development_value'
```
Note that in roles and templates you can now use `{{ some_variable }}`, which is a simple string.
When we use labeling with variables, you would use
```
cluster_type: production
```
or
```
cluster_type: development
```
in `group_vars/stack_name/vars.yml`, but you will then need a dict in `group_vars/all.yml` with something like this:
```
some_variable:
   production: 'production_value'
   development: 'development_value'
```
In roles and templates you must now use `{{ some_variable[cluster_type] }}` to lookup the correct value. Both strategies work and I have no strong preference for either one (yet)...